### PR TITLE
Add basic auth

### PR DESCRIPTION
**Pull request by mrwanny**
added a very basic http authentication filter so that malicious user does not have un-authentication access to perform destructive operation like deleting a topic.

thanks to https://github.com/Kaliber/play-basic-authentication-filter for the filter

### CODE REVIEW

File: app/controllers/BasicAuthenticationFilter.scala [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    def seq(key: String) =
+      Option(configuration.underlying getValue (root + key)).map { value =>
+        value.valueType match {
+          case ConfigValueType.LIST => value.unwrapped.asInstanceOf[java.util.List[String]].asScala
+          case ConfigValueType.STRING => Seq(value.unwrapped.asInstanceOf[String])
+          case _ => sys.error(s"Unexpected value at `${root + key}`, expected STRING or LIST")
+        }
+      }
+
+    val enabled = boolean("enabled").getOrElse(true)
```
Comment (patelh) [inline app/controllers/BasicAuthenticationFilter.scala]: Can we make this false by default instead of true.

Comment (mrwanny) [inline app/controllers/BasicAuthenticationFilter.scala]: It's actually set to false in the config file, but I can default it to false here too

**Comment by mrwanny** [comment]
I am not sere why test are failing on travis, locally things appears to work

**Comment by patelh** [comment]
please rebase of master

**Comment by mrwanny** [comment]
close and re-opening as master PR

**Comment by wadhwasam** [comment]
Can you send me step by step instructions how to enable basic authentication method.  I am new and still learning and need a small help from experts like you

**Comment by montana-ua** [comment]
@wadhwasam just open the config file. The answer is inside.

**Pull request closed without merging**